### ISSUE TITLE
[BUG] The Salt Minion is shutdown

### ISSUE THREAD

**Issue by max-arnold** [saltstack/salt#62741]
**Description**
A minion named `minion` can't be targeted:

**Setup**

```
root@master:~# salt --version
salt 3005

root@master:~# salt-key
Accepted Keys:
minion
Denied Keys:
Unaccepted Keys:
Rejected Keys:

root@master:~# salt minion test.ping
The Salt Minion is shutdown.

root@master:~# salt -L minion test.ping
minion:
    True
```

- [ x] onedir packaging
- [ x] used bootstrap to install

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
Salt Version:
          Salt: 3005

Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.0
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
        Python: 3.9.13 (main, Aug 23 2022, 18:31:04)
  python-gnupg: 0.4.8
        PyYAML: 5.4.1
         PyZMQ: 23.2.0
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-77-generic
        system: Linux
       version: Ubuntu 20.04 focal
```
</details>

**Comment by OrangeDog** [comment]
Does the regex (`-E`) matcher also fail?

What does `--preview-target` output if you use that instead of `test.ping`?

**Comment by max-arnold** [comment]
```
root@master:~# salt -E minion test.ping
minion:
    True

root@master:~# salt minion --preview-target
Usage: salt-minion [options]

salt-minion: error: no such option: --preview-target
```

I believe it is caused by the onedir binary parsing of positional arguments:

```
root@master:~# cat /usr/bin/salt
#!/bin/bash
exec /opt/saltstack/salt/run/run "${@:1}"
```

Targeting any of these minion names will fail:

```
root@master:~# /opt/saltstack/salt/run/run
Must pass in a salt command, available commands are:
minion
master
call
api
cloud
cp
extend
key
proxy
pip
run
shell
spm
ssh
support
syndic
python
```

```
root@master:~#  salt api test.ping
[INFO    ] Setting up the Salt API
[INFO    ] The salt-api is starting up
[INFO    ] Starting RunNetapi(salt.loaded.int.netapi.rest_cherrypy)
[INFO    ] Preparing the root key for local communication
[INFO    ] Preparing the root key for local communication
[INFO    ] Preparing the root key for local communication
[INFO    ] Preparing the root key for local communication
[INFO    ] Preparing the root key for local communication
[INFO    ] Preparing the root key for local communication
[INFO    ] Preparing the root key for local communication
[INFO    ] Preparing the root key for local communication
[INFO    ] [23/Sep/2022:19:06:51] ENGINE Listening for SIGTERM.
[INFO    ] [23/Sep/2022:19:06:51] ENGINE Listening for SIGHUP.
[INFO    ] [23/Sep/2022:19:06:51] ENGINE Listening for SIGUSR1.
[INFO    ] [23/Sep/2022:19:06:51] ENGINE Bus STARTING
[INFO    ] [23/Sep/2022:19:06:51] ENGINE Serving on http://0.0.0.0:8000
[INFO    ] [23/Sep/2022:19:06:51] ENGINE Bus STARTED
```

**Comment by OrangeDog** [comment]
https://docs.saltproject.io/en/latest/ref/cli/ has not been updated with any of that either.

**Comment by OrangeDog** [comment]
As there's no option to force glob matches, the best workaround is probably `-C`.

**Comment by dmurphy18** [comment]
Retired with onedir salt-master on Centos 8 Stream and salt-minion on Ubuntu 20.04 using onedir too, and able to reproduce the issue
```
root@Unknown:/home/david# salt-key
Accepted Keys:
Denied Keys:
Unaccepted Keys:
minion
Rejected Keys:
root@Unknown:/home/david# salt-key -y -a minion
The following keys are going to be accepted:
Unaccepted Keys:
minion
Key for minion minion accepted.
root@Unknown:/home/david# salt minion test.versions
^C[WARNING ] Minion received a SIGINT. Exiting.
The Salt Minion is shutdown. Minion received a SIGINT. Exited.
Minion received a SIGINT. Exiting.
The Salt Minion is shutdown. Minion received a SIGINT. Exited.
root@Unknown:/home/david# ps -ef | grep salt
root        2580       1  0 10:18 ?        00:00:03 /opt/saltstack/salt/run/run master MainProcess
root        2761    2580  0 10:18 ?        00:00:00 /opt/saltstack/salt/run/run master PubServerChannel._publish_daemon
root        2762    2580  0 10:18 ?        00:00:00 /opt/saltstack/salt/run/run master EventPublisher
root        2765    2580  0 10:18 ?        00:00:04 /opt/saltstack/salt/run/run master Maintenance
root        2766    2580  0 10:18 ?        00:00:00 /opt/saltstack/salt/run/run master ReqServer ReqServer_ProcessManager
root        2767    2766  0 10:18 ?        00:00:00 /opt/saltstack/salt/run/run master ReqServer MWorkerQueue
root        2770    2766  0 10:18 ?        00:00:03 /opt/saltstack/salt/run/run master ReqServer MWorker-0
root        2771    2766  0 10:18 ?        00:00:03 /opt/saltstack/salt/run/run master ReqServer MWorker-1
root        2772    2766  0 10:18 ?        00:00:03 /opt/saltstack/salt/run/run master ReqServer MWorker-2
root        2773    2766  0 10:18 ?        00:00:03 /opt/saltstack/salt/run/run master ReqServer MWorker-3
root        2774    2766  0 10:18 ?        00:00:03 /opt/saltstack/salt/run/run master ReqServer MWorker-4
root        5642    2580  0 10:23 ?        00:00:00 /opt/saltstack/salt/run/run master FileserverUpdate
root        6139    2437  0 10:28 pts/0    00:00:00 grep --color=auto salt
root@Unknown:/home/david# salt-key
Accepted Keys:
minion
Denied Keys:
Unaccepted Keys:
Rejected Keys:
root@Unknown:/home/david# salt \* test.versions
minion:
    Salt Version:
              Salt: 3005
     
    Dependency Versions:
              cffi: 1.14.6
          cherrypy: 18.6.1
          dateutil: 2.8.1
         docker-py: Not Installed
             gitdb: Not Installed
         gitpython: Not Installed
            Jinja2: 3.1.0
           libgit2: Not Installed
          M2Crypto: Not Installed
              Mako: Not Installed
           msgpack: 1.0.2
      msgpack-pure: Not Installed
      mysql-python: Not Installed
         pycparser: 2.21
          pycrypto: Not Installed
      pycryptodome: 3.9.8
            pygit2: Not Installed
            Python: 3.9.13 (main, Aug 23 2022, 18:31:04)
      python-gnupg: 0.4.8
            PyYAML: 5.4.1
             PyZMQ: 23.2.0
             smmap: Not Installed
           timelib: 0.2.4
           Tornado: 4.5.3
               ZMQ: 4.3.4
     
    System Versions:
              dist: ubuntu 20.04 focal
            locale: utf-8
           machine: x86_64
           release: 5.15.0-48-generic
            system: Linux
           version: Ubuntu 20.04 focal
     
root@Unknown:/home/david# salt minion test.versions
[ERROR   ] Error while bringing up minion for multi-master. Is master at salt responding?
^C[WARNING ] Minion received a SIGINT. Exiting.
The Salt Minion is shutdown. Minion received a SIGINT. Exited.
Minion received a SIGINT. Exiting.
The Salt Minion is shutdown. Minion received a SIGINT. Exited.
root@Unknown:/home/david# 
```

**Comment by dmurphy18** [comment]
@barbaricyawps We need to document reserved key words when using onedir for minion identifiers etc., see https://github.com/saltstack/salt/issues/62741#issuecomment-1256833979, and the reasons in the comment just above.

**Comment by dmurphy18** [comment]
The reserved key words issue should be resolved in the next major release of Salt, Sulfur (Sulphur).

**Comment by max-arnold** [comment]
Are there any technical ways to fix this (besides changing the docs)? Maybe by more smart shell wrapper scripts or argument parsing in run.py?

**Comment by dmurphy18** [comment]
@max-arnold The problem is position arguments and key word evaluation, implying making reserved key words out of minion, but didn't know the problem at the time, and given Tiamat based salt-minion have been around since 2019 (native minions). Even have testing with  **_minion_xxx_**,  so this is very much a corner case.
This is supposed to fixed in the next major release 3006.

**Comment by dmurphy18** [comment]
@max-arnold Checked this with the current development for 3006 (Sulfur) and the issue is resolved, but will wait for RC1 to be released, and recheck and close if still not an issue with Sulfur.

**Comment by garethgreenaway** [comment]
Closing this out, as this issue is resolved in the 3006.x branch and the fix will be available once 3006.0 has been released
